src/conversion/model-to-view-converters.js

```diff
@@ -307,17 +322,46 @@ export function move() {
  */
 export function remove() {
 	return ( evt, data, consumable, conversionApi ) => {
-		if ( consumable.consume( data.item, 'remove' ) ) {
-			const sourceModelRange = ModelRange.createFromPositionAndShift( data.sourcePosition, data.item.offsetSize );
-			const sourceViewRange = conversionApi.mapper.toViewRange( sourceModelRange );
+		if ( !consumable.consume( data.item, 'remove' ) ) {
+			return;
+		}
+
+		let viewRange = null;
 
-			viewWriter.remove( sourceViewRange );
+		if ( data.item instanceof ModelElement ) {
+			const viewElement = conversionApi.mapper.toViewElement( data.item );
+			viewRange = ViewRange.createOn( viewElement );
+		} else {
+			const viewPosition = conversionApi.mapper.toViewPosition( data.sourcePosition );
 
-			conversionApi.mapper.unbindModelElement( data.item );
+			viewRange = findViewTextRange( viewPosition, data.item.offsetSize );
 		}
+
+		viewWriter.remove( viewRange );
+		conversionApi.mapper.unbindModelElement( data.item );
 	};
 }
 
+// Helper function for `remove` and `move` converters. It returns a ViewRange that starts at ViewPosition `start` and
+// includes `size` characters.
+// This method is used to find a ViewRange basing on ModelPosition and ModelTextProxy item size in `move` and `remove`
+// converters where it is impossible to just map positions because those positions already are invalid in model
+// (because they got moved or removed).
```

Comment:
So that's fine for me. We can consider making `Range.createFromPositionAndShift` smarter in the future, but for now it is fine as it is.